tach/cache/__init__.py

```diff
@@ -0,0 +1,5 @@
+from __future__ import annotations
```

Comment:
Does this need to be here? I haven't used future statements before but maybe it's better to add it only to files that use annotations, or if it applies once globally at import time, I think we can put it in `tach/__init__.py`

Reply:
It's a requirement as of #76 

```
[tool.ruff.lint.isort]
required-imports = ["from __future__ import annotations"]
```

Reply:
It's just easier to have it everyone from my experience, some big codebases like https://github.com/apache/airflow do this.